code/game/objects/effects/anomalies/anomaly_bleed.dm

```diff
@@ -0,0 +1,60 @@
+/obj/effect/anomaly/blood
+	name = "blood anomaly"
+	icon_state = "blood_anomaly"
+	density = TRUE
+	aSignal = /obj/item/assembly/signaler/anomaly/blood
+	var/sucking = FALSE
+
+/obj/effect/anomaly/blood/anomalyEffect(delta_time)
+	if (sucking)
+		return
+	..()
+	if (DT_PROB(20, delta_time))
+		AddComponent(/datum/component/pellet_cloud, projectile_type=/obj/projectile/bullet/shrapnel/bleed, magnitude=3)
+		playsound(src, 'sound/weapons/shrapnel.ogg', 70, TRUE)
+	if (DT_PROB(10, delta_time))
+		for (var/mob/living/carbon/bleed_target in view(6, src))
+			if (!bleed_target.resting)
+				continue
+			if (do_teleport(src, get_turf(bleed_target), channel = TELEPORT_CHANNEL_BLUESPACE))
+				bleed_target.Stun(3 SECONDS)
+				INVOKE_ASYNC(src, PROC_REF(suck_blood))
+			break
+
+/obj/effect/anomaly/blood/detonate()
+	STOP_PROCESSING(SSobj, src)
```

Comment:
It shouldn't belong here, because admin can detonate it forcefully, but it will still remained here.
![image](https://github.com/user-attachments/assets/f447f7d2-f7e0-4ad4-869a-9058ff5c8e44)
but also it has it already. not necessary.

Reply:
This is necessary since it sleeps. If the processing wasn't stopped then it would stop moving before it detonates. If an admin detonates it forcefully, then it will be deleted at the end of its process and will work as normal (but also if an admin is calling procs in a dodgy and unintended way then its not really a bug)

Reply:
then I think leaving a comment why it is a thing there would be nice since it's really unusual.